gtda/diagrams/features.py

```diff
@@ -126,7 +127,10 @@ def fit(self, X, y=None):
         validate_params(
             self.get_params(), self._hyperparameters, exclude=['n_jobs'])
 
-        self.homology_dimensions_ = sorted(set(X[0, :, 2]))
+        self.homology_dimensions_ = tuple(
+            sorted([int(dim) if dim != np.inf else dim
+                    for dim in set(X[0, :, 2])])
+            )
```

Comment:
This logic appears a total of 7 times in the code. It's pretty simple but I what would you say about extracting it to a common function `homology_dimensions_to_ints`, in `gtda.diagrams._utils`?

Reply:
Yeah, that could be done!

Reply:
Implemented in https://github.com/giotto-ai/giotto-tda/pull/454/commits/c2367aa3420702b08421f99285555a0307c4f657, could you have a look?

Reply:
Yes!
Oh, I just realized :facepalm: that we look for dimensions only in the first diagram! What about something like `set(np.ravel(X[:,:,2]))`?

Reply:
This is by design, as my comments and the docstrings explain. I think it's correct like this, unless I'm missing something.

Reply:
I am about to suggest a change to `_subdiagrams` to be even more informative to the user about this.

Reply:
Yes - good point!